src/main/java/uk/gov/hmcts/cmc/claimstore/repositories/CCDCaseApi.java

```diff
@@ -65,22 +74,103 @@ public CCDCaseApi(
     }
 
     public Optional<Claim> getByExternalId(String externalId, String authorisation) {
-        final List<Claim> claims = search(authorisation, ImmutableMap.of("case.externalId", externalId));
+        return getCaseDetailsByExternalId(authorisation, externalId)
+            .map(CaseDetails::getData)
+            .map(this::convertToCCDCase)
+            .map((caseMapper::from));
+    }
 
-        if (claims.size() > 1) {
+    public Optional<Claim> linkDefendant(String externalId, String defendantId, String authorisation) {
+        User user = userService.authenticateAnonymousCaseWorker();
+        Optional<CaseDetails> optionalCaseDetails = getCaseDetailsByExternalId(authorisation, externalId);
+
+        if (optionalCaseDetails.isPresent()) {
+            CaseDetails caseDetails = optionalCaseDetails.get();
+            caseAccessApi.grantAccessToCase(user.getAuthorisation(),
+                authTokenGenerator.generate(),
+                user.getUserDetails().getId(),
+                JURISDICTION_ID,
+                CASE_TYPE_ID,
+                caseDetails.getId().toString(),
+                new UserId(defendantId)
+            );
+
+            return Optional.of(caseMapper.from(convertToCCDCase(caseDetails.getData())));
+        }
+        return Optional.empty();
```

Comment:
We just silently return a blank optional if linking fails because given case was not found. I think we should throw an exception in such case.

Reply:
up a level it throws a not found exception

Reply:
I still believe throwing the exception is the responsibility of this method.

Reply:
You think a NotFoundException should come from here?
Happy to do but normally would throw a bit higher.